src/tiled/objectselectiontool.cpp

```diff
@@ -551,8 +586,25 @@ void ObjectSelectionTool::mouseReleased(QGraphicsSceneMouseEvent *event)
             break;
         }
         const Qt::KeyboardModifiers modifiers = event->modifiers();
+        QSet<MapObjectItem*> selection = mapScene()->selectedObjectItems();
+        if (modifiers & Qt::AltModifier) {
+            auto underlyingObjects = objectItemsAt(event->scenePos());
+
+            if (underlyingObjects.isEmpty())
+                break;
+            auto currRotation = std::find(underlyingObjects.begin(),
+                                          underlyingObjects.end(), mSelectedInUnderlyingRotation);
```

Comment:
Please prefer to use Qt methods when available. This should just be `underlyingObjects.indexOf(mSelectedInUnderlyingRotation)`.

However, I suggested to not use any other state than the current selection, to make this action completely predictable. There should not be a `mSelectedInUnderlyingRotation`, but rather you could search for the bottom-most selected object and then select the next one (cyling back to top-most object if it doesn't exist).

It seems reverse iterators have only been added to QList in Qt 5.6, and Tiled currently needs to compile with Qt 5.4, so I think there will be not such a short way of finding the bottom-most selected object. It would probably just be easiest to do something like:
```c++
MapObjectItem *nextItem = underlyingObjects.first();
for (int i = underlyingObjects.size() - 1; i > 0; --i) {
    MapObjectItem *underlyingObject = underlyingObjects.at(i);
    if (selection.contains(underlyingObject)
        break;
    nextItem = underlyingObject;
}
```